### PR TITLE
Add Windows support for storage monitor

### PR DESCRIPTION
**Pull request by ark0f**

### CODE REVIEW

**Review by bkchr — APPROVED** [review]

**Review by michalkucharczyk — APPROVED** [review]

**Comment by nazar-pc** [comment]
I'd like to request faster upstreaming here, without this `sc-service` is not possible to compile on Windows either.

P.S. Would be great to have basic `cargo c` on Windows platform in CI.

**Comment by bkchr** [comment]
> I'd like to request faster upstreaming here, without this `sc-service` is not possible to compile on Windows either.
> 
> P.S. Would be great to have basic `cargo c` on Windows platform in CI.

We are waiting for @ark0f to accept the proposed changes and then we can merge it. We don't have permissions to commit to his branch.

File: client/storage-monitor/src/lib.rs [hunk 1]
```diff
@@ -117,12 +118,8 @@ impl StorageMonitorService {
...
 	/// Returns free space in MB, or error if statvfs failed.
 	fn free_space(path: &Path) -> Result<u64, Error> {
-		statvfs(path)
-			.map(|stats| {
-				u64::from(stats.blocks_available()).saturating_mul(u64::from(stats.block_size())) /
-					1_000_000
-			})
-			.map_err(Error::from)
+		let amount = statvfs(path)?;
+		Ok(amount.available_space() / (1024 * 1024))
```
[the remark this answers is not in the document]
Comment (ark0f) [reply client/storage-monitor/src/lib.rs:122]: Why do we divide by 1 million here?

[the remark this answers is not in the document]
Comment (bkchr) [reply client/storage-monitor/src/lib.rs:122]: As the comment says it is MB and not MiB

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2457341

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2457342

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2457347

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2457340

**Comment by ark0f** [comment]
Oops. I'm sorry. Rebased on wrong master branch

**Comment by ggwpez** [comment]
Please dont force push. You can just do `git merge origin/master` and then push that.

**Review by ggwpez — APPROVED** [review]

**Comment by ggwpez** [comment]
bot merge

**Pull request merged**